improver/cli/threshold.py

```diff
@@ -140,29 +140,32 @@ def process(
         thresholds = [np.float32(x) for x in threshold_values]
         fuzzy_bounds = None
 
-    result_no_collapse_coord = BasicThreshold(
+    each_threshold_func_list = []
+
+    if vicinity is not None:
+        # smooth thresholded occurrences over local vicinity
+        each_threshold_func_list.append(OccurrenceWithinVicinity(vicinity))
+
+    if collapse_coord is not None:
+        # Take a weighted mean across realizations with equal weights
+        each_threshold_func_list.append(
+            WeightAndBlend(collapse_coord, "linear", y0val=1.0, ynval=1.0)
+        )
```

Comment:
Order of operations is important here, but that's not distinguished inside the plugin as far as I can see.  Can we add some checks to the plugin to make sure that if both of these methods are set, they are called in the right order?

Reply:
the order of operations is fixed, i.e. the order of list append, so `OccurrenceWithinVicinity` goes always before `WeightAndBlend` if both are applied. I think there's no need to test it.

Reply:
OK, having checked again this is covered in a CLI test, so agree there's no need for further testing.